### ISSUE TITLE
ARIA for fields in core Blockly

### ISSUE THREAD

**Issue by rachel-fenichel** [google/blockly#8205]
### Check for duplicates

- [X] I have searched for similar issues before opening a new one.

### Problem

ARIA defines roles, states, and properties. These expose information to a browser's accessibility API, which sets which information is exposed to the user through their screen reader when they move to an individual DOM element.

This issue does not include:
- Keyboard navigation (how to move from one field to another)
- "Announcing" (describing where an element is within the larger context of the workspace or page)

### Request

For each field in core Blockly:
- Apply the correct [ARIA roles and properties](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques) to the DOM elements that represent the field
- Provide [accessible names and descriptions](https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/) to the DOM elements that represent the field
- Make sure the field works both during editing and during display (not editing)
- Use the correct keyboard shortcuts to move into and out of editing mode when on a single instance of the field (e.g. enter, space, and esc)

- [ ] checkbox
- [ ] dropdown
- [ ] image
- [ ] text input
- [ ] label
- [ ] serializable label
- [ ] number input
- [ ] variable

In addition, provide APIs as needed for custom fields to support shared keyboard shortcuts (enter, space, escape).

### Additional context

This is a parent issue for ARIA in fields. Information related to individual fields, such as discussion of the correct behaviour for a specific field, should be tracked in sub-issues.

Information related to all fields can be tracked here.

**Comment by rachel-fenichel** [comment]
Any field with a validator may require [input assistance](https://www.w3.org/WAI/WCAG22/Understanding/input-assistance.html). This should be as consistent as possible across all of our fields, so it may need to be implemented at a higher level instead of per-field.
